### ISSUE TITLE
TypeError: argument must be a dict, not _DialectArgView when running sqlalchemy

### ISSUE THREAD

**Issue by xqzhou** [tonybaloney/Pyjion#28]
```
(.venv) admin-mbp:python-macrobenchmarks admin$ python data/flaskblogging_serve.py
Traceback (most recent call last):
  File "/Users/admin/Workspace/python/python-macrobenchmarks/data/flaskblogging_serve.py", line 24, in <module>
    sql_storage = SQLAStorage(engine, metadata=meta)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/flask_blogging/sqlastorage.py", line 68, in __init__
    self._metadata.reflect(bind=self._engine)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 4511, in reflect
    Table(name, self, **reflect_opts)
  File "<string>", line 2, in __new__
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py", line 139, in warned
    return fn(*args, **kwargs)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 560, in __new__
    metadata._remove_table(name, schema)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.raise_(
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 555, in __new__
    table._init(name, metadata, *args, **kw)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 654, in _init
    self._init_items(*args)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/sql/schema.py", line 667, in _autoload
    autoload_with.run_callable(
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1653, in run_callable
    return callable_(self, *args, **kwargs)
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 469, in reflecttable
    return insp.reflecttable(
  File "/Users/admin/Workspace/python/pyjion/.venv/lib/python3.9/site-packages/sqlalchemy/engine/reflection.py", line 647, in reflecttable
    tbl_opts = self.get_table_options(
TypeError: argument must be a dict, not _DialectArgView
(.venv) admin-mbp:python-macrobenchmarks admin$
```

**Comment by tonybaloney** [comment]
Would have been raised by https://github.com/tonybaloney/Pyjion/blob/master/src/pyjion/intrins.cpp#L1089-L1112

Assume that `_DialectArgView` has a dict-protocol implementation.

Think the fix would be to replace `PyDict_CheckExact` with `PyDict_Check`

**Comment by tonybaloney** [comment]
Further research shows that `_DialectArgView` is a `collections.MutableMapping` type and implements the `Mapping` protocol

**Comment by tonybaloney** [comment]
will be released in 0.0.6

**Comment by tonybaloney** [comment]
Resolved in 0.0.7
